crates/orchestrator/src/jobs/da_job/mod.rs

```diff
@@ -396,25 +396,33 @@ fn da_word(class_flag: bool, nonce_change: Option<Felt>, num_changes: u64) -> Fe
 }
 
 fn refactor_state_update(state_update: &mut StateDiff) {
```

Comment:
i think it can still be made better per say

1. create `existing_storage_addresses` on the top
2. loop over nonce, if not present in `existing_storage_addresses`, insert in `state_update` and `existing_storage_addresses`
3. same for deployed_contracts

So, 1 hash set and 2 loops. Complexity remains same to be precise but it's cleaner. `required_addresses` feels redundant here.

Reply:
Ig one loop is cleaner. the above method seems same only.